receiver/hostmetricsreceiver/internal/scraper/cpuscraper/internal/cpu_utilization_calculator_test.go

```diff
@@ -0,0 +1,288 @@
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func TestCpuUtilizationCalculator_Calculate(t *testing.T) {
+
+	testCases := []struct {
+		name                 string
+		now                  pdata.Timestamp
+		previousTime         pdata.Timestamp
+		cpuTimes             []cpu.TimesStat
+		previousCPUTimes     []cpu.TimesStat
+		expectedUtilizations []CPUUtilization
+		expectedError        error
+	}{
+		{
+			name:         "no previous times",
+			previousTime: 1640097430772858000,
+			now:          1640097435776827000,
+			cpuTimes: []cpu.TimesStat{
+				{
+					CPU:  "cpu0",
+					User: 8260.4,
+				},
+			},
+		},
+		{
+			name:         "no delta time",
+			previousTime: 1640097430772858000,
+			now:          1640097430772858000,
+			previousCPUTimes: []cpu.TimesStat{
+				{
+					CPU:  "cpu0",
+					User: 8259.4,
+				},
+			},
+			cpuTimes: []cpu.TimesStat{
+				{
+					CPU:  "cpu0",
+					User: 8260.4,
+				},
+			},
+			expectedError: fmt.Errorf("invalid elapsedSeconds: %f", 0.0),
+		},
+		{
+			name:         "invalid TimesStats",
+			previousTime: 1640097430772858000,
+			now:          1640097430772858000,
+			previousCPUTimes: []cpu.TimesStat{
+				{
+					CPU:  "cpu5",
+					User: 8259.4,
+				},
+			},
+			cpuTimes: []cpu.TimesStat{
+				{
+					CPU:  "cpu6",
+					User: 8260.4,
+				},
+			},
+			expectedError: errors.New("cannot find TimesStat for cpu: cpu5"),
+		},
+		{
+			name:         "one cpu",
+			previousTime: 1640097430772858000,
+			now:          1640097435776827000,
+			previousCPUTimes: []cpu.TimesStat{
+				{
+					CPU:    "cpu0",
+					User:   8258.4,
+					System: 6193.3,
+					Idle:   34284.7,
+				},
+			},
+			cpuTimes: []cpu.TimesStat{
+				{
+					CPU:    "cpu0",
+					User:   8259.4,
+					System: 6193.9,
+					Idle:   34288.2,
+				},
+			},
+			expectedUtilizations: []CPUUtilization{
+				{
+					CPU:    "cpu0",
+					User:   0.19984,
+					System: 0.11990,
+					Idle:   0.69944,
+				},
+			},
+		},
+		{
+			name:         "multiple cpus unordered",
+			previousTime: 1640097430772858000,
+			now:          1640097435776827000,
+			previousCPUTimes: []cpu.TimesStat{
+				{
+					CPU:    "cpu1",
+					User:   528.3,
+					System: 549.7,
+					Idle:   47638.2,
+				},
+				{
+					CPU:    "cpu0",
+					User:   8258.4,
+					System: 6193.3,
+					Idle:   34284.7,
+				},
+			},
+			cpuTimes: []cpu.TimesStat{
+				{
+					CPU:    "cpu0",
+					User:   8259.4,
+					System: 6193.9,
+					Idle:   34288.2,
+				},
+				{
+					CPU:    "cpu1",
+					User:   528.4,
+					System: 549.7,
+					Idle:   47643.1,
+				},
+			},
+			expectedUtilizations: []CPUUtilization{
+				{
+					CPU:    "cpu1",
+					User:   0.01998,
+					System: 0,
+					Idle:   0.97922,
+				},
+				{
+					CPU:    "cpu0",
+					User:   0.19984,
+					System: 0.11990,
+					Idle:   0.69944,
+				},
+			},
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
If test is thread-safe:
```suggestion
	for _, test := range testCases {
		t.Run(test.name, func(t *testing.T) {
		test := test
		t.Parallel()
```

Reply:
👍  didn't know about this one thanks! :) but I saw that it should be
```
for _, test := range testCases {
   test := test
   t.Run(test.name, func(t *testing.T) {
   t.Parallel()
```
changing